### DISCUSSION TITLE
MJCF and Mujoco Rendering

### DISCUSSION THREAD

**Discussion by nsk126** [google-deepmind/mujoco#1512] [Asking for Help] [answered]
I've been meaning to use the procedural manipulation from MJCF for my Mujoco models and simualtion, But I've been having difficulty integrating the two.

For example, the `MjModel` and `MjData` instance we use for simulation and rendering (in `mujoco.viewer`) seem to be different from the `physics.model` and `physics.data` that is generated using `dm_control.mjcf`

```python
mjcf_model = mjcf.from_path(<xml_path>)
physics = mjcf.Physics.from_mjcf_model(mjcf_model)

with viewer.launch_passive(
        model=physics.model,
        data=physics.data
    ) as viewer:
```

Results in an error:
```
model is not a mujoco.MjModel: got dm_control.mujoco.wrapper.core.MjModel object.
```

Does anyone have experience working with them together and can guide me / point to resouces? Would really appreciate it! Thanks!

**Comment by kevinzakka** [comment, marked as answer]
Change it to physics.data.ptr and physics.model.ptr.

**Reply by nsk126** [reply]
Works! Much appreciated!
